src/hotspot/share/ci/ciMethod.cpp

```diff
@@ -1161,17 +1161,12 @@ bool ciMethod::was_executed_more_than(int times) {
 // ------------------------------------------------------------------
 // ciMethod::has_unloaded_classes_in_signature
 bool ciMethod::has_unloaded_classes_in_signature() {
-  VM_ENTRY_MARK;
-  {
-    ExceptionMark em(THREAD);
-    methodHandle m(THREAD, get_Method());
-    bool has_unloaded = Method::has_unloaded_classes_in_signature(m, thread);
-    if( HAS_PENDING_EXCEPTION ) {
-      CLEAR_PENDING_EXCEPTION;
-      return true;     // Declare that we may have unloaded classes
+  for (ciSignatureStream str(signature()); !str.is_done(); str.next()) {
```

Comment:
Add comment here why we don't go into VM for the answer to avoid confusion in a future.

Reply:
I slightly reshuffled the code and added comments. Does it look clearer now?